src/app/core/services/external-cluster.ts

```diff
@@ -244,6 +254,42 @@ export class ExternalClusterService {
     return this._http.post<ExternalCluster>(url, externalClusterModel, {headers}).pipe(catchError(() => of<null>()));
   }
 
+  deleteExternalCluster(projectID: string, clusterID: string, action: string): Observable<void> {
+    const url = `${this._newRestRoot}/projects/${projectID}/kubernetes/clusters/${clusterID}`;
+    const headers = new HttpHeaders({
+      Action: action,
+    });
+    return this._http.delete<void>(url, {headers: headers});
+  }
+
+  showDisconnectClusterDialog(cluster: ExternalCluster, projectID: string): void {
+    const dialogConfig: MatDialogConfig = {
+      data: {
+        title: 'Disconnect Cluster',
+        message: `Are you sure you want to disconnect <b>${cluster.name}</b> cluster?`,
+        confirmLabel: 'Disconnect',
+        throttleButton: true,
+        observable: this._disconnectExternalCluster(projectID, cluster.id),
+      },
+    };
+
+    this._matDialog
+      .open(ConfirmationDialogComponent, dialogConfig)
+      .afterClosed()
+      .pipe(filter(isConfirmed => isConfirmed))
+      .subscribe(_ => {
+        this._router.navigate(['/projects/' + projectID + '/clusters'], {
+          fragment: `${ClusterListTab.ExternalCluster}`,
+        });
+        this._notificationService.success(`Disconnected the ${cluster.name} cluster`);
+      });
+  }
+
+  private _disconnectExternalCluster(projectID: string, clusterID: string): Observable<void> {
+    const url = `${this._newRestRoot}/projects/${projectID}/kubernetes/clusters/${clusterID}`;
```

Comment:
Looks like this is same API endpoint that is used in `deleteExternalCluster` method. Can we somehow avoid this duplication? Perhaps create common method for this API request and add `headers` parameter in that method?

Reply:
should i update on same PR?